phoenix-core/src/main/java/org/apache/phoenix/schema/TableRef.java

```diff
@@ -111,8 +115,10 @@ public boolean equals(Object obj) {
         if (obj == null) return false;
         if (getClass() != obj.getClass()) return false;
         TableRef other = (TableRef)obj;
+        // a null alias on either side should mean a wildcard and should not fail the equals check
```

Comment:
I don't think we are really implementing what is stated by this comment. We should remove it. 
Also, the alias needs a similar check the way it is for table.

if (!Objects.equal(alias, other.alias)) return false.